src/System.Windows.Forms/src/System/Windows/Forms/Button.cs

```diff
@@ -399,5 +389,43 @@ protected override void WndProc(ref Message m)
                     break;
             }
         }
+
+        protected override AccessibleObject CreateAccessibilityInstance()
+           => new ButtonAccessibleObject(this);
+
+        [ComVisible(true)]
+        internal class ButtonAccessibleObject : ControlAccessibleObject
+        {
+            internal ButtonAccessibleObject(Button owner) : base(owner)
+            {
+            }
+
+            public override string Name
+            {
+                get => base.Name ?? SR.ButtonDefaultAccessibleName;
```

Comment:
Could you please specify the case when we need `SR.ButtonDefaultAccessibleName` to be set as accessible name? By the current convention we should not return control type name as accessible name as in this case Narrator will announce "Button button" where the first "Button" is accessible name and the second "button" is accessible control type. Can we remove the override Name property from here? Does it break anything?

Reply:
I used ProgressBarAccessibleObject as an example. So because I still not fully understand our project I just implemented it like in ProgressBarAccObj